spec/controllers/prior_authority/primary_quote_controller_spec.rb

```diff
@@ -1,149 +1,31 @@
-              } }
-      end
-
-      it 'redirects back to form' do
-        expect(response).to redirect_to(edit_prior_authority_steps_primary_quote_path(application))
-      end
-
-      it 'generates flash error' do
-        expect(flash[:alert])
-          .to eq('File potentially contains malware so cannot be uploaded. Please contact your administrator')
-      end
-    end
-
-    context 'when incorrect file type is uploaded' do
-      before do
-        request.env['CONTENT_TYPE'] = 'application/json'
-        put :update,
-            params: { application_id: '12345',
-            prior_authority_steps_primary_quote_form:
-            {
-              document: fixture_file_upload('test.png', 'application/json')
-            } }
-      end
-
-      it 'redirects back to form' do
-        expect(response).to redirect_to(edit_prior_authority_steps_primary_quote_path(application))
-      end
-
-      it 'generates flash error' do
-        expect(flash[:alert])
-          .to eq('The selected file must be a DOC, DOCX, XLSX, XLS, RTF, ODT, JPG, BMP, PNG, TIF or PDF')
-      end
-    end
-
-    context 'when file size too big' do
-      before do
-        request.env['CONTENT_TYPE'] = 'image/png'
-        put :update,
-            params: { application_id: '12345',
-            prior_authority_steps_primary_quote_form:
-            {
-              document: fixture_file_upload('test_2.png', 'image/png')
-            } }
-      end
-
-      it 'redirects back to form' do
-        expect(response).to redirect_to(edit_prior_authority_steps_primary_quote_path(application))
-      end
-
-      it 'generates flash error' do
-        expect(flash[:alert])
-          .to eq('The selected file must be smaller than 10MB')
-      end
-    end
-
     context 'when there is a standard error' do
-      before do
-        request.env['CONTENT_TYPE'] = 'image/png'
-        allow(controller).to receive(:upload_file).and_raise(StandardError)
-        put :update,
-            params: { application_id: '12345',
-            prior_authority_steps_primary_quote_form:
-            {
-              document: fixture_file_upload('test.png', 'image/png')
-            } }
-      end
+      let(:form_object) { instance_double(PriorAuthority::Steps::PrimaryQuoteForm) }
 
-      it 'redirects back to form' do
-        expect(response).to redirect_to(edit_prior_authority_steps_primary_quote_path(application))
+      before do
+        allow(PriorAuthority::Steps::PrimaryQuoteForm).to receive(:new).and_return(form_object)
+        allow(form_object).to receive(:save).and_raise(StandardError)
+        allow(form_object).to receive_message_chain(:errors, :add)
       end
 
-      it 'generates flash error' do
-        expect(flash[:alert])
-          .to eq('Unable to upload file at this time')
+      it 're-shows the form with an error message' do
+        expect(form_object).to receive_message_chain(:errors, :add)
```

Comment:
**nit** could test the error message itself here

```
# warning, have not tried this not sure it will work with receive_message_chain
expect(form_object).to receive_message_chain(:errors, :add).with(:base, 'my super informative error message')
```

Reply:
I _think_ the i18n doesn't occur until you pull the message _out_ of `errors`, but you're right, it's still worth testing. Will fix.